### PR TITLE
Individual row collapsed in case review

### PR DESCRIPTION
**Pull request by hanars**
Moves the behavior for toggling whether a FamilyLayout` component is collapsed or not into a helper component instead of implementing directly for Family table rows, so behavior can be easily shared for individual rows. This does require some tweaking for the display, but overall everything (except case review) looks the same

### CODE REVIEW

File: ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  static propTypes = {
+    layoutComponent: PropTypes.elementType.isRequired,
+    detailFields: PropTypes.arrayOf(PropTypes.object).isRequired,
+    noDetailFields: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  state = { showDetails: false }
+
+  toggle = () => {
+    this.setState(prevState => ({ showDetails: !prevState.showDetails }))
```
Comment (ShifaSZ) [inline ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx:15]: Using an object parameter for the `setState` function should be sufficient.

File: ui/shared/components/panel/family/Family.jsx [hunk 2]
```diff
@@ -189,15 +190,15 @@ class Family extends React.PureComponent {
               {`(${family.individualGuids.length})`}
             </span>
           ) : (
-            <div key="header">
+            <span key="header">
```
Comment (ShifaSZ) [inline ui/shared/components/panel/family/Family.jsx:193]: Is the `key` prop necessary?

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx (hunk 1 above)
Comment (hanars) [reply ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx:15]: I don't understand what that means. This is adapted from a copy-paste

File: ui/shared/components/panel/family/Family.jsx (hunk 2 above)
Comment (hanars) [reply ui/shared/components/panel/family/Family.jsx:193]: yes

File: ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx (hunk 1 above)
Comment (ShifaSZ) [reply ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx:15]: Using a function for the setState parameter allows access to the `latest` state value after previous calls to the `setState`. We don't have multiple calls to the `setState` before re-rendering. So it is safe and neater to use an object parameter like `this.setSate({ showDetails: !prevState.showDetails })`.

Comment (hanars) [reply ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx:15]: thats a violation of our linter and less stable than the function approach used here

**Pull request merged**